src/plan.rs

```diff
@@ -73,6 +73,52 @@ const MAX_RADIX4_BITS: u32 = 16; // largest size to consider radix 4 an option i
 const MAX_RADER_PRIME_FACTOR: usize = 23; // don't use Raders if the inner fft length has prime factor larger than this
 const MIN_BLUESTEIN_MIXED_RADIX_LEN: usize = 90; // only use mixed radix for the inner fft of Bluestein if length is larger than this
 
+#[derive(Debug, std::cmp::PartialEq, Clone)]
+#[allow(dead_code)]
+pub enum Recipe {
+    DFT(usize),
+    MixedRadix {
+        left_fft: Box<Recipe>,
```

Comment:
What do you think about changing this to `Rc`? It seems like as a part of the planner's recipe cache, we expect to be cloning recipes a lot. For example, `design_fft_with_factors()` calls clone() on the recipe when returning it from the cache. This will trigger a deep copy of potentially very many boxes. If it was a Rc instead, then the clone in `design_fft_with_factors()` would just increment a refcount.

Reply:
Good idea! That should save a little time, and I don't see any disadvantages. I'll change and update.

Reply:
This is just a thought, but since we're caching recipes now, we could update `plan_fft()` to always call `design_fft()`, and then always pass that recipe into `build_fft()`.

The benefit would be that the plan method can be significantly simpler, and the only drawback would be that planning a ftt that's already planned would be 2 hashtable lookups instead of one. seems worth it imo. what do you think?

Reply:
Both suggestions implemented! I think the simplification is well worth one extra hashmap lookup.